platform-api/src/main/java/io/datacater/core/deployment/DeploymentSpec.java

```diff
@@ -5,18 +5,48 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import java.util.Map;
+import java.util.UUID;
 
-public record DeploymentSpec(Map<String, String> deployment) {
-  public static JsonNode serializeDeploymentSpec(Map<String, String> deployment)
-      throws JsonProcessingException {
+public class DeploymentSpec {
+  @JsonProperty("pipeline-in")
+  private final String pipelineIn;
+
+  @JsonProperty("pipeline-out")
+  private final String pipelineOut;
```

Comment:
Would it make sense to rename `pipeline-{in,out}` to `stream-{in,out}`? The current name sounds like a deployment connects with two pipelines.

Also, we should reference the UUIDs of the streams.

Reply:
> Would it make sense to rename `pipeline-{in,out}` to `stream-{in,out}`? The current name sounds like a deployment connects with two pipelines.

Thanks for catching this, it actually needs to be stream instead of pipelin (the config at least, since thats the naming in pipeline image)

 
> Also, we should reference the UUIDs of the streams.

I guess we need to rethink this. Right now we're referencing the stream names and using it when setting up a deployment. But we don't actually need to reference either here? We reference stream-in/out UUID in pipeline and the streamNames are found in stream object. I guess it depends on where we want our source of truth, inside deployment or inside pipeline.

My opinion: We could remove deployment-spec all together. User only need to enter pipeline UUID when creating a deployment, from there we get all the info we need. The only thing we would need is a deployment name -> can use pipeline name

Reply:
Yes, makes sense 👍 Let's remove this information from the deployment and use the config from the pipeline spec.

When passing the configuration to Kubernetes, would it make sense to "inline" the stream config in the config map?

Reply:
> When passing the configuration to Kubernetes, would it make sense to "inline" the stream config in the config map?

you mean with configmap instead of passing environment variables?

Reply:
Yeah, since we do not know all config options for sure, it might be more flexible. What do you think about it?

Reply:
Smallrye sadly isn't that flexible as far as i know. It requires stream-in and stream-out config to be set upon startup.

Reply:
there seems to be two options for configuring our streams in deployment:

1. We map everything separately in the environment variables
2. We use [kafka-config](https://quarkus.io/guides/kafka#kafka-configuration-resolution)
    * We can either add one environment variable with a map, and these are mapped through `@Identifier("my-configuration")`-method
    * We add a configmap with map these with `@Identifier("my-configuration")`-method

Reply:
I've updated deployment and pipeline to now forward the kafka-config to our pipeline.
`boostrap.servers` and `key/value.(de)serializer` are take from `StreamSpec`, but will be overwritten if these values are found in the `DeploymentSpec`


```
{
  "pipeline": "dc161a69-fa49-4b1a-b1b1-6d8246d50d72",
  "stream-in-config": {
    "bootstrap.servers": "localhost:9092",
    ....
  },
  "stream-out-config": {
    "bootstrap.servers": "localhost:9092",
    .....
  }
}
```
is the format we are looking for in deploymentSpec, where `stream-in-config` and `stream-out-config` are optional